### ISSUE TITLE
Z-wave smartplug with thermostat (4512725) looses connection

### ISSUE THREAD

**Issue by paaland** [easyhomesrc/homey_namronas#71]
I have four of these. They work for about 3-6 hours, then they stop sending data and stop responding to commands (all are in switch mode). Even the on/off button on the device stops working. Only way to get them back is to disconnect power. Then they work for a few hours again.

They are all connected to heating elements (panelovn) of 1000 W.

Latest Homey Pro firmware and latest app version.

**Comment by maginawin** [comment]
Hi @paaland ,

Thanks for your reply, I'm about to test this issue now.

**Comment by paaland** [comment]
I've done some more testing and pulled down the code locally. I did notice that the zwave security settings were missing from app.json for the 4512725 device. I added that locally and pushed the version to my Homey. I also removed all four devices and only added one with the modified app. It has been running for 12 hours now and is still sending data and responding to commands. I'll leave it running some more, then I'll add another device. If this is pure luck, if it has something to do with having multiple vs single device or the missing security settings fixed anything is too early to tell. 

The device is paired with S0 security by the way.

```
"security": [
          "none",
          "S0"
        ],
```

Section above is missing in app.json for the 4512725 device.

**Comment by maginawin** [comment]
Hi @paaland ,

Thanks for your reply.

I set all Z-Wave security to `'security': ['none', 'S0'],` to solve the problem with S2, which you can see here on line 23.

https://github.com/easyhomesrc/homey_namronas/blob/master/drivers/4512725/driver.compose.json

Did you add this device to Homey before No 26, 2021? If so, they may be added as S2 devices.

**Comment by paaland** [comment]
Hi. Can't remember the date I added them, but I remember the zwave page on
delevoper.athom.com stating S0.

Pål Andreassen
e: ***@***.***
m: (+47) 982 85 504

ons. 29. des. 2021, 11:00 skrev Maginawin ***@***.***>:

**Comment by paaland** [comment]
Nope. Now the device is offline again and not responding to homey. Was
online for 9-10 hours this time.

Since even the on/off switch on the device stops working I'm leaning
towards a firmware issue, that the thing itself has crashed. Any way to
firmware upgrade these things?

Pål Andreassen
e: ***@***.***
m: (+47) 982 85 504

ons. 29. des. 2021, 11:17 skrev Pål Andreassen ***@***.***>:

**Comment by maginawin** [comment]
Hi @paaland ,

I also think the device have crashed. If you don't add your device to Homey, will it stop working too?

I guess Homey doesn't support firmware upgrades, I didn't find the SDK documentation.

**Comment by paaland** [comment]
The other three devices (not paired with Homey) seems to still work. In that I can press the button on them to turn the switch on and off. Only the one connected to Homey seems to have "crashed".

Seems I'll be contacting Elektroimportøren and ask for a refund. To bad, I had hope for these devices.

**Comment by maginawin** [comment]
Hi @paaland ,

I'm sorry about that. My device still works after 24 hours, I use Homey (Early 2019) and Homey version is 7.1.6. The screenshot below for details for the device.

![IMG_7765](https://user-images.githubusercontent.com/8154155/147802675-19d0db99-9e62-4945-abdc-76552d61e367.PNG)
